File: src/frontend/src/lib/riskColors.ts
import type { RiskTier } from "../api/types";

// Single source of truth for risk-tier color values, mirroring the Tailwind
// tokens defined in src/index.css (--color-risk-*). RiskBadge uses the
// Tailwind utility classes directly; anything that needs a raw CSS/hex value
// (map markers, inline styles, canvas/SVG) should import this instead of
// re-declaring the palette.
export const RISK_TIER_HEX: Record<RiskTier, string> = {
  Critical: "#9c2b1f",
  High: "#a85f17",
  Medium: "#8f6c0c",
  Low: "#2c6b46",
};

export const RISK_TIER_ORDER: RiskTier[] = ["Critical", "High", "Medium", "Low"];

File: src/mobile/lib/util/risk_colors.dart
import 'package:flutter/material.dart';

import '../api/types.dart';
import '../theme/app_theme.dart';

/// Single source of truth for risk-tier color values, mirroring the web
/// frontend's `src/lib/riskColors.ts` and the `--color-risk-*` tokens in
/// `src/index.css`.
const Map<RiskTier, Color> riskTierColor = {
  RiskTier.critical: AppColors.riskCritical,
  RiskTier.high: AppColors.riskHigh,
  RiskTier.medium: AppColors.riskMedium,
  RiskTier.low: AppColors.riskLow,
};

const Map<RiskTier, Color> riskTierBgColor = {
  RiskTier.critical: AppColors.riskCriticalBg,
  RiskTier.high: AppColors.riskHighBg,
  RiskTier.medium: AppColors.riskMediumBg,
  RiskTier.low: AppColors.riskLowBg,
};

const List<RiskTier> riskTierOrder = [
  RiskTier.critical,
  RiskTier.high,
  RiskTier.medium,
  RiskTier.low,
];

Color colorForTier(RiskTier tier) => riskTierColor[tier] ?? AppColors.riskLow;
